main.tf

```diff
@@ -12,9 +12,18 @@ provider "aws" {
   region = "us-east-1"
 }
 
+variable "env_code" {
```

Comment:
I see that you already have the environemnt_tag variable which is very similar.
We can re-use it instead of introducing a new one